src/lib/operators/export.cpp

```diff
@@ -10,7 +10,10 @@
 namespace opossum {
 
 Export::Export(const std::shared_ptr<const AbstractOperator>& in, const std::string& filename, const FileType& type)
-    : AbstractReadOnlyOperator(OperatorType::Export, in), _filename(filename), _type(type) {}
+    : AbstractReadOnlyOperator(OperatorType::Export, in), _filename(filename), _type(type) {

```

Comment:
I do not know whether it is better to use an input operator here or, to be kind of align with the ImportOperator, only provide a table name, which is then accessed from the storage manager in the ::on_execute(). This, for example, has an influence whether we need to check for the table existence in `sql_pipeline_statement.cpp` (when we provide a table name) or in `lqp_translator.cpp` or even in `sql_translator.cpp` (when we use an input operator).

Reply:
Is there any reason not to use an input operator?

Reply:
Using a table name aligns with DropTable, DropView etc. that also access data maintained by the StorageManager. Checking if this data (i.e. table) exists can then also be done in SQLPipelineStatement, where it is done for the other operators.

Reply:
> that also access data maintained by the StorageManager

So does pretty much everyone else.

The big benefit of making this an operator (which I just thought of now) is that it respects the MVCC visiblity of rows.